nssp_demo/forecast_state.py

```diff
@@ -0,0 +1,222 @@
+import argparse
+import logging
+import os
+import subprocess
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import numpyro
+import polars as pl
+from prep_data import process_and_save_state
+
+numpyro.set_host_device_count(4)
```

Comment:
Consider reading this from the command line or setting to something very high by default.

Reply:
In general should figure out the best way to coordinate parallelization settings.